internal/integration/kafka/kafka.go

```diff
@@ -52,10 +53,24 @@ func New(m marshaler.Type, conf config.IntegrationKafkaConfig) (*Integration, er
 	}
 
 	if conf.Username != "" || conf.Password != "" {
-		wc.Dialer.SASLMechanism = plain.Mechanism{
-			Username: conf.Username,
-			Password: conf.Password,
+		switch conf.Mechanism {
```

Comment:
I think it is good to add a `default` which errors that an unknown kafka authentication mechanism was configured, to make sure there is an error when the config value is not set to `plain` or `scram`.